cos-fleetshard-sync/src/main/java/org/bf2/cos/fleetshard/sync/resources/cleanup/AddonConfigMapWatcher.java

```diff
@@ -0,0 +1,91 @@
+package org.bf2.cos.fleetshard.sync.resources.cleanup;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.Watch;
+import org.bf2.cos.fleetshard.api.ManagedConnector;
+import org.bf2.cos.fleetshard.api.ManagedConnectorOperator;
+import org.bf2.cos.fleetshard.support.resources.Resources;
+import org.bf2.cos.fleetshard.support.watch.AbstractWatcher;
+import org.bf2.cos.fleetshard.sync.FleetShardSync;
+import org.bf2.cos.fleetshard.sync.FleetShardSyncConfig;
+import org.bf2.cos.fleetshard.sync.client.FleetShardClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+
+public class AddonConfigMapWatcher extends AbstractWatcher {
```

Comment:
I would put this as part of the resources sync task instead of using a watcher as in case something goes wrong, then there won't be any new configmap creation event  so we may leave leftovers

Reply:
I understand we need a way to not have leftovers if something fails, but I don't get how we could do this in resources sync task. We need to execute this as soon as someone tries to delete the addon, and and that is precisely when this ConfigMap is created.

Reply:
I've made some changes. See if this is more inline with what you meant.